### PR TITLE
604 tests for sync status

### LINKED ISSUE #604: Tests for sync status
Tests to check sync status is logged correctly during synchronisation

### PR DESCRIPTION
**Pull request by andreievg**
closes #604 

I am not entirely happy about the test itself, it's somewhat lengthy and not as simple as I wanted. Couldn't really validate that statuses are logged correctly without mocking central server and going through the whole synchronisation (and asserting status within mocked central server routes).

Did quite a few changes on top of actual test:

* Relocate sync status and logger to common directory, the actually change after relocation is this: https://github.com/openmsupply/open-msupply/compare/d6ba48f9cf34ad286223a1e977f2cf8db628885e..3290861e80fe8cec053c9e33ce1a6e994fce5186
* Added batch_size to sync settings rather then having hard coded constants, this is needed for sync log test, when doing push we want to see progress for more then one batch, and with const 1024 batch size would have to create too many records. (also updated batch size for integration tests to be on a small size, to test iterations)
* Changed `progress_start` to `progress_total` in schema
* `progress` method takes one parameter, `remaining` and sets progress_total and progress_done for the step appropriately
*  Logging to console is moved from synchroniser logic, to logger
* Added tests and updated push and pulls to align with test results
* After merge from feature/sync-v5, renamed service

### CODE REVIEW

File: server/service/src/sync/central_data_synchroniser.rs [hunk 1]
```diff
@@ -66,18 +62,18 @@ impl CentralDataSynchroniser {
...
-                    sync_batch.max_cursor as u64,
-                )
+                .progress(SyncStepProgress::PullCentral, max_cursor - cursor)
                 .map_err(SyncLoggerError)?;
 
-            if batch_length == 0 {
-                info!("Central sync buffer is up-to-date");
-                break;
+            match (batch_length, cursor < max_cursor) {
+                (0, false) => break,
```
Comment (mark-prins) [inline server/service/src/sync/central_data_synchroniser.rs:70]: haven't seen the match with tuples before! that's very neat

File: server/service/src/sync/sync_status/status.rs [hunk 2]
```diff
@@ -80,23 +96,23 @@ pub fn get_latest_sync_status(
...
         pull_remote_progress_done,
         integration_start_datetime,
         integration_done_datetime,
         error_message,
-    } = SyncLogRowRepository::new(&connection).load_latest_sync_log()?;
+    } = SyncLogRowRepository::new(&ctx.connection).load_latest_sync_log()?;
 
     let result = FullSyncStatus {
-        is_syncing: done_datetime.is_some() || error_message.is_some(),
+        is_syncing: done_datetime.is_none() && error_message.is_none(),
```
Comment (mark-prins) [inline server/service/src/sync/sync_status/status.rs:115]: :nit I'm getting confused between the usage of `done` as a counter of completed items and `done_datetime` which is, I think, completed datetime. the word `done` is being a bit flexible here

File: server/service/src/sync/sync_status/status.rs [hunk 3]
```diff
@@ -113,20 +129,20 @@ pub fn get_latest_sync_status(
         pull_central: pull_central_start_datetime.map(|started| SyncStatusWithProgress {
             started,
             finished: pull_central_done_datetime,
-            total_progress: pull_central_progress_start.map(i32_to_u32).unwrap_or(0),
-            done_progress: pull_central_progress_done.unwrap_or(0) as u32,
+            total_progress: pull_central_progress_total.map(i32_to_u32),
```
Comment (mark-prins) [inline server/service/src/sync/sync_status/status.rs:132]: :nit `total_progress`? isn't this just `total` 
I don't see what the progress adds to the name

**Review by mark-prins — APPROVED** [review]
👍  Looks good to me!

I've formatted, as some of the lines were obviously non-standardly formatted and that's blown out the number of files a bit 🙄 

The test runs fine though - haven't checked against actual sync, will wait for the UI integration

File: server/service/src/sync/sync_status/status.rs (hunk 2 above)
Comment (andreievg) [reply server/service/src/sync/sync_status/status.rs:115]: How about ?

in database:
start_datetime -> started_datetime
done_datetime -> finished_datetime 

in graphql: 
total_progress -> total
done_progress -> done (in graphql can comment in schema, which will be `done`)

Comment (andreievg) [reply server/service/src/sync/sync_status/status.rs]: 291d72aa0f3f96df67c9a944ad6b066c42ffe148

File: server/service/src/sync/sync_status/status.rs (hunk 3 above)
Comment (andreievg) [reply server/service/src/sync/sync_status/status.rs]: 291d72aa0f3f96df67c9a944ad6b066c42ffe148

**Comment by andreievg** [comment]
🙏 fo review

**Pull request merged**